lang/src/main/java/org/kaazing/k3po/lang/internal/ast/AstConnectNode.java

```diff
@@ -93,4 +94,20 @@ public String getBarrier() {
     public void setBarrier(String barrier) {
         this.barrier = barrier;
     }
+
+    public AstValue getLocation() {
+        return location;
+    }
+
+    public void setLocation(AstValue location) {
```

Comment:
per current design, location can either take - 
- expression value (AstExpressionValue extends AstValue)
- location literal (AstLocationLiteral extends AstLocation extends AstValue)

That's why location is of type AstValue.